fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -318,6 +323,71 @@ private void updateDimensionRow(
     }
 
     /**
+     * Replaces Lucene Index with a new index.
+     *
+     * @param newLuceneIndexPath  The path of the new index
+     */
+    public void replaceIndex(String newLuceneIndexPath) {
+        Directory oldLuceneDirectory = luceneDirectory;
+
+        lock.writeLock().lock();
+        try {
+            // move the old dir to a different location
+            try {
+                Files.move(
+                        new File(luceneIndexPath).getAbsoluteFile().toPath(),
+                        new File(renameToTemp(luceneIndexPath)).getAbsoluteFile().toPath(),
+                        REPLACE_EXISTING
+                );
+            } catch (IOException e) {
+                String message = ErrorMessageFormat.CANNOT_MOVE_INDEX_DIR.format(luceneIndexPath);
+                LOG.error(message, e);
+            }
+
+            // open a new lucene index dir
+            try {
+                luceneDirectory = new MMapDirectory(Paths.get(newLuceneIndexPath));
+            } catch (IOException e) {
+                String message = ErrorMessageFormat.CANNOT_CREATE_INDEX_DIR.format(newLuceneIndexPath);
+                LOG.error(message, e);
+            }
+
+            luceneIndexPath = newLuceneIndexPath; // have lucene point to the new lucene index dir
+
```

Comment:
This still doesn't work.  We need the new files moved to the same location as the old ones to make this resilient across restarts.